### PR TITLE
fix: install requirements.txt in writable dir

### PR DESCRIPTION
**Pull request by jesterhazy**
*Issue #, if available:*

- python service fails to install requirements.txt

*Description of changes:*

fixes for: 

- python service fails to install requirements.txt
- other issues with python lib handling
- other issues with python service startup
- request error handling and error responses
- failure to shutdown gunicorn
- add gevent to improve performance
- fix defective tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://3r4crpxqvk.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=7f5d9a12-f042-4fc4-9a3a-065327496d2a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: container/sagemaker/python_service.py [hunk 1]
```diff
@@ -31,13 +37,9 @@
...
 
     def __init__(self):
-        if 'TFS_GRPC_PORT' not in os.environ:
-            raise EnvironmentError('GRPC port not set')
-        if 'TFS_REST_PORT' not in os.environ:
-            raise EnvironmentError('REST API port not set.')
-
-        self._tfs_grpc_port = os.getenv('TFS_GRPC_PORT')
-        self._tfs_rest_port = os.getenv('TFS_REST_PORT')
+        self._tfs_default_model_name = os.environ['TFS_DEFAULT_MODEL_NAME']
```
Comment (ChoiByungWook) [inline container/sagemaker/python_service.py]: Should these have defaults?

Comment (jesterhazy) [reply container/sagemaker/python_service.py]: these do have defaults in `serve.py` so they should always be set when `python_service.py` is run. defaulting them again would probably just hide a configuration error.

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://3r4crpxqvk.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4ac8234e-4258-47f4-8a6f-93d962c32b37%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**